Library/Homebrew/software_spec.rb

```diff
@@ -534,6 +533,14 @@ def sha256(hash)
     end
 
     cellar ||= all_tags_cellar
+    cellar ||= if tag.to_s.end_with?("_linux")
+      Homebrew::DEFAULT_LINUX_CELLAR
+    elsif tag.to_s.start_with?("arm64_")
+      Homebrew::DEFAULT_MACOS_ARM_CELLAR
```

Comment:
I'd love to avoid the `_linux`/`arm64` string usage here or at least be able to consolidate other usages into a function rather than having these strings scattered throughout the codebase. Note at least the `arm64_` case could be handled by `MacOS::Version.arch`.

Reply:
> Note at least the `arm64_` case could be handled by `MacOS::Version.arch`.

Given the format of arm64 macOS tags are identical to Linux, perhaps we should have a more generic `Utils::Bottle::Tag` class that does the arch stuff that `MacOS::Version` has been doing, but in a more generic OS manner.